ui/components/app/transaction-list/transaction-list.component.js

```diff
@@ -334,13 +334,64 @@ export default function TransactionList({
     setSelectedTransaction(transaction);
   }, []);
 
+  const getTransactionDisplayAmount = (transaction, userAddress) => {
+    const userFromEntry = transaction.from?.find(
+      (entry) => entry.address === userAddress,
+    );
+
+    const userToEntry = transaction.to?.find(
+      (entry) => entry.address === userAddress,
+    );
+
+    // Amount of the token sent
+    if (userFromEntry?.asset?.amount) {
+      return `-${userFromEntry.asset.amount} ${userFromEntry.asset.unit || ''}`;
+    }
+
+    // Amount of the token received
+    if (userToEntry?.asset?.amount) {
+      return `${userToEntry.asset.amount} ${userToEntry.asset.unit || ''}`;
+    }
+
+    // Fallback: Amount of the token received
+    if (transaction.from?.[0]?.asset?.amount) {
+      return `${transaction.from?.[0]?.asset?.amount} ${
+        transaction.from?.[0]?.asset?.unit || ''
+      }`;
+    }
+
+    return '';
+  };
+
   const multichainNetwork = useMultichainSelector(
     getMultichainNetwork,
     selectedAccount,
   );
 
   const trackEvent = useContext(MetaMetricsContext);
 
+  const formatTransactionTitle = (transaction, userAddress) => {
+    switch (transaction.type) {
+      case TransactionType.swap: {
+        const userToEntry = transaction.to?.find(
+          (entry) => entry.address === userAddress,
+        );
+        const userFromEntry = transaction.from?.find(
+          (entry) => entry.address === userAddress,
+        );
+
+        if (userFromEntry && userToEntry) {
```

Comment:
Is there a reason why we check if the user is in both `from` and `to` since we know from the transaction type that it's a swpa?

Reply:
Just because `.find` can return `null`

Reply:
I suppose you only need the `userToEntry`. Maybe this logic can be improved in a different PR (it can be a problem for other blockchains, we can discuss it later)